### PR TITLE
feat: support custom url on image workers

### PR DESCRIPTION
**Pull request by preschian**
- [x] Related https://github.com/kodadot/nft-gallery/issues/6618

## Background
Some of nft creators put metadata in non-ipfs, for example, this one https://kodadot.xyz/stmn/collection/88

it will return a CORS error if we just hit that endpoint directly

image

## Features
in this PR, extend the possibility to image workers to support custom URL

usage:
```
http://127.0.0.1:8787/type/url?endpoint=<url-here>
```

for example:
```
// will store in our r2 bucket
http://127.0.0.1:8787/type/url?endpoint=https://polkadot-data.s3.us-east-2.amazonaws.com/metadata/nfts-88/nfts-88_collection-meta.json

// will store in our cf-images
http://127.0.0.1:8787/type/url?endpoint=https://polkadot-data.s3.us-east-2.amazonaws.com/metadata/nfts-88/nfts-88_collection-img.png
```

### CODE REVIEW

**Comment by vikiival** [comment]
Btw images can be uploaded via url ;)

**Comment by vikiival** [comment]
> > Btw images can be uploaded via url ;)
> 
> 
> 
> Which "images" part do you mean?

True images word is ambigous in the context. 

https://developers.cloudflare.com/images/cloudflare-images/upload-images/upload-via-url/

**Comment by preschian** [comment]
> [developers.cloudflare.com/images/cloudflare-images/upload-images/upload-via-url](https://developers.cloudflare.com/images/cloudflare-images/upload-images/upload-via-url/)

actually, that one is the same thing. since the worker is using javascript, this is to upload image to the cf-images

https://github.com/kodadot/workers/blob/cb6c5d78e20b58dc64b24bb098c163ee0a9c5d3f/image/src/utils/cloudflare-images.ts#L35-L45

if you visit this page https://developers.cloudflare.com/api/operations/cloudflare-images-upload-an-image-via-url, you can choose between cURL or javascript fetch on the top right side
image

**Comment by preschian** [comment]
oopps sorry, I just understand what you mean

yes, already use URL to upload that. you can see it here

https://github.com/kodadot/workers/blob/cb6c5d78e20b58dc64b24bb098c163ee0a9c5d3f/image/src/utils/cloudflare-images.ts#L31-L33

File: image/package.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   "version": "0.0.0",
-  "devDependencies": {
-    "@cloudflare/workers-types": "^3.18.0",
-    "typescript": "^4.9.4",
-    "wrangler": "^2.6.2"
-  },
   "private": true,
   "scripts": {
-    "start": "wrangler dev",
+    "start": "wrangler dev --remote",
```
Comment (vikiival) [inline image/package.json:6]: maybe add option for dev?

**Review by vikiival — CHANGES_REQUESTED** [review]
Can we rename `CFI` to Image?

File: image/package.json (hunk 1 above)
Comment (preschian) [reply image/package.json]: updated

**Comment by preschian** [comment]
> Can we rename `CFI` to Image?

I feel like `Image` is too general. Our worker name is also `image`

**Comment by preschian** [comment]
I will deploy this to canary env, to test it on our UI

**Comment by preschian** [comment]
@vikiival @roiLeo I will merge this, lmk if you had a concern regarding this PR

**Pull request merged**